client/src/components/Form/Elements/FormDialog.vue

```diff
@@ -0,0 +1,68 @@
+<script setup lang="ts">
+import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
+import { library } from "@fortawesome/fontawesome-svg-core";
+import { faFolderOpen } from "@fortawesome/free-solid-svg-icons";
+import DataDialog from "@/components/DataDialog/DataDialog.vue";
+import { onMounted } from "vue";
+import { getCurrentGalaxyHistory } from "@/utils/data";
+import { reactive } from "vue";
+
+library.add(faFolderOpen);
+
+interface DataDialogProps {
+    id: string;
+    multiple?: boolean;
+    value?: Array<string> | string;
+}
+
+const props = withDefaults(defineProps<DataDialogProps>(), {
+    multiple: false,
+});
+
+const title = "Browse Datasets";
+const historyID: string = "";
+const localHistoryId = reactive({
+    value: historyID,
+});
+
+onMounted(() => {
+    getCurrentGalaxyHistory().then((historyId) => {
+        localHistoryId.value = historyId;
+    });
+});
+
+const initialODD = false;
+const openDataDialog = reactive({
+    value: initialODD,
+});
+
+const dataSelected = reactive({
+    value: props.value,
+});
+
+const emit = defineEmits<{
+    (e: "input", value: Array<string> | string): void;
+}>();
+
+function onData(result: Array<string> | string) {
+    openDataDialog.value = false;
+    dataSelected.value = result;
+    emit("input", result);
+}
+</script>
+
+<template>
+    <div class="d-flex">
+        <span @click="openDataDialog.value = true">
```

Comment:
Please do not add click handlers to non-interactive elements, as this breaks accessibility. Instead, use a `<button>` element